subprojects/packagefiles/glfw/tests/meson.build

```diff
@@ -0,0 +1,133 @@
+###  GLFW test programs
+#@ Build check
+if get_option('build-tests') == false
+  subdir_done()
+endif
+
+#@ Constants
+tst_include = include_directories('../include', '../src', '../deps')
+glad_gl = '../deps/glad_gl.c'
+glad_vulkan = '../deps/glad_vulkan.c'
+getopt = '../deps/getopt.c'
+tinycthread = '../deps/tinycthread.c'
+
+#@ Flags
+if is_posix == true
+  c_flags = ['-D_XOPEN_SOURCE=700']
+endif
+
+#@ Test executables
+tests = [['clipboard', [getopt, glad_gl], tst_clipboard],
+         ['events', [getopt, glad_gl], tst_events],
+         ['msaa', [getopt, glad_gl], tst_msaa],
+         ['glfwinfo', [getopt, glad_gl, glad_vulkan], tst_glfwinfo],
+         ['iconify', [getopt, glad_gl], tst_iconify],
+         ['monitors', [getopt, glad_gl], tst_monitors],
+         ['reopen', glad_gl, tst_reopen],
+         ['cursor', glad_gl, tst_cursor],
+         ['empty', [tinycthread, glad_gl], tst_empty],
+         ['gamma', glad_gl, tst_gamma],
+         ['icon', glad_gl, tst_icon],
+         ['inputlag', [getopt, glad_gl], tst_inputlag],
+         ['joysticks', glad_gl, tst_joysticks],
+         ['opacity', glad_gl, tst_opacity],
+         ['tearing', glad_gl, tst_tearing],
+         ['threads', [tinycthread, glad_gl], tst_threads],
+         ['timeout', glad_gl, tst_timeout],
+         ['title', glad_gl, tst_title],
+         ['triangle-vulkan', glad_vulkan, tst_trianglevulkan],
+         ['windows', [getopt, glad_gl], tst_windows]]
+foreach t : tests
+  name = t[0]
+  src = [t[0] + '.c'] + t[1]
+  t[2] = executable(name, src,
```

Comment:
The lexer doesn't like this: "Assignment target must be an id."

You cannot indirectly assign *to* an array element like this, but that being said, the array contains nonexistent variables.

I would instead suggest assigning it to `exe =`, then defining the test immediately inside the loop as well.